zip-0312.rst

```diff
@@ -1,7 +1,466 @@
+spend authorization private key :math:`\mathsf{ask}` [#protocol-spendauthsig]_
+is the particular key that must be used in the context of this ZIP. Note that
+while Sapling allows creating it directly, in Orchard it is always derived
+from the spending key :math:`\mathsf{ask}`. This means that a trusted
+dealer key generation process might be required as detailed in [#frost-tdkg]_.
+
+
+Randomizer Generation
+'''''''''''''''''''''
+
+A new helper function is defined, which computes :math:`\mathsf{RedDSA.GenRandom}`:
+
+::
+
+  randomizer_generate():
+
+  Inputs:
+  - None
+
+  Outputs: randomizer, a Scalar
+
+  def randomizer_generate():
+    randomizer_input = random_bytes(64)
+    return H3(randomizer_input)
+
+
+Binding Factor Computation
+''''''''''''''''''''''''''
+
+The `compute_binding_factor` function is changed to receive the `randomizer_point`
+as follows: ::
+
+  Inputs:
+  - commitment_list = [(i, hiding_nonce_commitment_i, binding_nonce_commitment_i), ...],
+    a list of commitments issued by each participant, where each element in the list
+    indicates a NonZeroScalar identifier i and two commitment Element values
+    (hiding_nonce_commitment_i, binding_nonce_commitment_i). This list MUST be sorted
+    in ascending order by identifier.
+  - msg, the message to be signed.
+  - randomizer_point, an element in G.
+
+  Outputs:
+  - binding_factor_list, a list of (NonZeroScalar, Scalar) tuples representing the binding factors.
+
+  def compute_binding_factors(commitment_list, msg, randomizer_point):
+    msg_hash = H4(msg)
+    encoded_commitment_hash = H5(encode_group_commitment_list(commitment_list))
+    rho_input_prefix = msg_hash || encoded_commitment_hash
+
+    binding_factor_list = []
+    for (identifier, hiding_nonce_commitment, binding_nonce_commitment) in commitment_list:
+      rho_input = rho_input_prefix || G.SerializeScalar(identifier)
+      binding_factor = H1(rho_input)
+      binding_factor_list.append((identifier, binding_factor))
+    return binding_factor_list
+
+
+Round One - Commitment
+''''''''''''''''''''''
+
+Roune One is exactly the same as specified #[FROST]_. But for context, it
+involves these steps:
+
+- Each signer generates nonces and their corresponding public commitments.
+  A nonce is a pair of Scalar values, and a commitment is a pair of Element values.
+- The nonces are stored locally by the signer and kept private for use in the second round.
+- The commitments are sent to the Coordinator.
+
+
+Round Two - Signature Share Generation
+''''''''''''''''''''''''''''''''''''''
+
+In Round Two, the Coordinator generates a random scalar `randomizer` by calling
+`randomizer_generate`. Then it computes `randomizer_point = G.ScalarBaseMult(randomizer)`
+and sends it to each signer, over a confidential and authenticated channel,
+along with the message and the set of signing commitments. (Note that this differs
+from regular FROST which just requires an authenticated channel.)
```

Comment:
In Zcash the message being signed is just the sighash. How do the signers get sufficient information to understand what payment they are authorizing?

Note that the pre-signed transaction is not sufficient either; the signers need at least an opening of the value commitment correspondimg to the amount they're authorizing to be spent, and a decryption of the note ciphertexts so that they know where the funds are being sent to.

It could be argued that the details are outside the scope of this ZIP (since that will be defined by the protocol used by a given hardware wallet or similar authorizing device) but pointing out the security requirement is definitely in scope.